src/com/spike/Expr.java

```diff
@@ -52,23 +52,6 @@
     }
   }
 
-  static class Logical extends Expr {
```

Comment:
If you ask why I have deleted it, it is because it is fully identical to `Expr.Binary`, and also you haven't used it in parser so this code is unneeded.